src/capture.rs

```diff
@@ -359,66 +359,65 @@ pub mod source {
         let token_mut = &mut token;
         messages_op.build(move |capabilities| {
 
-            // Vector of strong references to capabilities, which can be dropped to terminate the sources.
+            // A Weak that communicates whether the returned token has been dropped.
             let drop_activator_weak = Arc::downgrade(&drop_activator);
 
             *token_mut = Some(drop_activator);
 
             // Read messages from some source; shuffle them to UPDATES and PROGRESS; share capability with FEEDBACK.
-            // First, wrap capabilities in a rc refcell so that they can be downgraded to weak references.
-            let capability_sets = (CapabilitySet::from_elem(capabilities[0].clone()), CapabilitySet::from_elem(capabilities[1].clone()));
-            let capability_sets = ActivateOnDrop::new(capability_sets, Rc::new(address), activations);
-            let mut local_capabilities = Some(capability_sets);
+            let mut capability_sets = (CapabilitySet::from_elem(capabilities[0].clone()), CapabilitySet::from_elem(capabilities[1].clone()));
```

Comment:
Nit, but can I trick you into breaking this out as `update_caps` and `progress_caps`?